boot/bootutil/src/loader.c

```diff
@@ -48,6 +48,18 @@ void sim_assert(int x);
 
 static struct boot_loader_state boot_data;
 
+#ifdef MCUBOOT_VALIDATE_SLOT0
```

Comment:
I admit I'm nervous about the correctness of this patch. No specific issues, it's just making code that's already a bit complex even more so and have more ifdefs.

I think that any failures here are probably a sign of bad hardware, and that mcuboot in general should fail fast, especially in critical code like this part.

Is there a use case driving this?

Reply:
> Is there a use case driving this?

E.g. if it's just to work around write failures caused randomly by 75037fd ("Add configurable simulation of bad memory writes"), I think that either the simulation should disable those errors in certain areas that "shouldn't" fail, or the errors should be taken seriously here.

Reply:
> I think that any failures here are probably a sign of bad hardware, and that mcuboot in general should fail fast, especially in critical code like this part.
>
> Is there a use case driving this?

This was an agreed on behavior to use in a meeting and thankfully (for my fuzzy memory!) it was written down here: https://runtimeco.atlassian.net/browse/MCUB-31

```
We will examine the code and make sure: 
If it is a data write failure then panic 
If it is a status failure then ignore
```
Also I am pretty sure this came from someone not from Runtime, @d3zd3z might remember the rationale.

> E.g. if it's just to work around write failures caused randomly by 75037fd ("Add configurable simulation of bad memory writes"), I think that either the simulation should disable those errors in certain areas that "shouldn't" fail, or the errors should be taken seriously here.

I think this is exactly what it does! The random fails are configured by address/size, so I leave it disabled for most tests apart from the ones (two if remember correctly) that I want to inspect behavior of failure handling. And for those I only mark as bad area the status area inside the trailer leaving most of the flash memory unaffected and behaving as fail on every error.

Reply:
> This was an agreed on behavior to use in a meeting and thankfully (for my fuzzy memory!) it was written down here: https://runtimeco.atlassian.net/browse/MCUB-31

Alright, if this has been discussed and agreed, I'm OK with being nervous in my little corner instead ;). Thanks for the pointer.